tardis/tests/tests_slow/conftest.py

```diff
@@ -39,6 +42,53 @@ def pytest_unconfigure(config):
     shutil.rmtree(config.option.tempdir)
 
 
+@pytest.mark.hookwrapper
+def pytest_runtest_makereport(item, call):
+    # execute all other hooks to obtain the report object
+    outcome = yield
+    report = outcome.get_result()
+    plot_object = getattr(item, "plot_object", None)
```

Comment:
This hookwrapper get's executed for every test.
You should check whether the test has an associated `plot_obj` or not. Only if there is a `plot_obj` a plot should be uploaded and the report may be modified.

I suggest to put all the logic inside the PlotUploader and the to something like `report.extra = plot_obj.get_extra()`.